modules/octo-sts-secret/main.tf

```
@@ -0,0 +1,55 @@
+// Create a dedicated GSA for the rotating GitHub tokens.
+resource "google_service_account" "octo-sts-rotator" {
+  project = var.project_id
+
+  account_id   = var.name
+  display_name = "Octo STS Secret Rotator"
+  description  = "Dedicated service account for rotating the Octo STS token for ${var.name}"
+}
+
+module "gh-token-secret" {
+  source = "../secret"
+
+  project_id = var.project_id
+  name       = var.name
+
+  service-account  = var.service_account
+  authorized-adder = "serviceAccount:${google_service_account.octo-sts-rotator.email}"
+
+  notification-channels = var.notification_channels
+}
+
+// The cron rotation job needs to list & delete access to cleanup old secrets.
+resource "google_secret_manager_secret_iam_binding" "authorize-manage" {
+  secret_id = module.gh-token-secret.secret_id
+  role      = "roles/secretmanager.secretVersionManager"
+  members   = ["serviceAccount:${google_service_account.octo-sts-rotator.email}"]
+}
+
+module "this" {
+  source = "../cron"
+
+  name       = var.name
+  project_id = var.project_id
+  region     = var.region
+
+  invokers = var.invokers
+
+  timeout = "60s" // 1 minute
+  # Run every 30 minutes
+  schedule = "*/30 * * * *"
```

```suggestion
  schedule = "17-59/30 * * * *"
```

nit: add an offset

https://stackoverflow.com/questions/12786410/run-cron-job-every-n-minutes-plus-offset